src/protocolsupport/protocol/typeremapper/itemstack/complex/ItemStackComplexRemapper.java

```diff
@@ -6,6 +6,7 @@
 @FunctionalInterface
 public interface ItemStackComplexRemapper {
 
+	//TODO: All of the item remappers return a NetworkItemStack, but never make a new NetworkItemStack. they're always mutated
```

Comment:
That is because they might actually require one itemstackremapper do something first. (Like remove special name enchantments etc. It is designed to be applied on top of eachother.)